src/htm/algorithms/TemporalMemory.hpp

```diff
@@ -677,13 +677,17 @@ class TemporalMemory : public Serializable
       Real anomaly_ = 0.5f; //default value
       ANMode mode_ = ANMode::RAW;
       AnomalyLikelihood anomalyLikelihood_; //TODO provide default/customizable params here
-  };
+      void reset() {
+        anomaly_ = 0.5f;
+	mode_ = ANMode::RAW;
+	//TODO provide anomalyLikelihood_.reset();
```

Comment:
when calling `setAnomalyMode` during a running sequence, the likelihood score would be broken, should `reset`. But reset is not available now for Likelihood (and I don't intend to implement it in this PR, so it stays broken).